modules/smtp/modules.php

```diff
@@ -1576,7 +1576,7 @@ public function process() {
 
         foreach ($servers as $server_id => $config) {
             $mailbox = new Hm_Mailbox($server_id, $this->user_config, $this->session, $config);
-            if ($mailbox->connect()) {
+            if ($mailbox || $mailbox->connect()) {
```

Comment:
Can you leave this line as it was because the $mailbox cannot empty in this case ?